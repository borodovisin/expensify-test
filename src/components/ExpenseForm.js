import React, { Component } from 'react';

export default class ExpenseForm extends Component {
    constructor(props){ 
        super(props);
        this.state = {
            description: props.expense ? props.expense.description : '',
            note: props.expense ? props.expense.note : '',
            amount: props.expense ? (props.expense.amount / 100).toString() : '',
            error: ''
        }
    }

    onDescriptionChanges = e => {
        const description =  e.target.value;
        this.setState(() => ({ description }));
    }

    onNoteChanges = e => {
        const note =  e.target.value;
        this.setState(() => ({ note }));
    }

    onAmountChanges = e => {
        const amount = e.target.value;
        if (!amount ||  amount.match(/^\d*(\.\d{0,2})?$/)) this.setState(() => ({amount}));
    }

    onSubmit = e => {
        e.preventDefault();

        if (!this.state.description || !this.state.amount) {
            this.setState(() => ({ error: "Description and amount are mandatory fields" }));
        } else {
            this.setState(() => ({ error: '' }));
            this.props.onSubmit({
                description: this.state.description,
                amount: parseFloat(this.state.amount, 10) * 100,
                note: this.state.note
            });
        }
    }

    render() {
        return (
            <div>
                <form onSubmit={this.onSubmit}>
                    <input type="text" placeholder="Description" autoFocus value={this.state.description}
                        onChange={this.onDescriptionChanges} />
                    <input type="text" placeholder="Amount" value={this.state.amount} 
                    onChange={this.onAmountChanges}/>
                    <textarea placeholder="Add a note for your expense (optional)" onChange={this.onNoteChanges} value={this.state.note} />
                    <button>Add Expense</button>
                    {this.state.error && <p>{this.state.error}</p>}
                </form>
            </div>
        );
    }
}