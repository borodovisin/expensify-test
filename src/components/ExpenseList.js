import React from 'react';
import { connect } from 'react-redux';

import ExpenseListItem from './ExpenseiveListIten';

export const ExpenseList = (props) => (
    <div>
        <h1>Expense List</h1>
        {props.expenses.map(expense => {
            return <ExpenseListItem key={expense.id} {...expense} />
        })}
    </div>
);

const mapStateProps = state => {
    return {
        expenses: state.expenses,
        filters: state.filters
    }
}

export default connect(mapStateProps)(ExpenseList);