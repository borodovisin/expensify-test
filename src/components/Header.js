import React from 'react';
import { NavLink, Link } from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify-</h1>
        <ul>
            <li><NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink></li>
            <li><NavLink to="/create" activeClassName="is-active" exact={true}>Create Expense</NavLink></li>
            <li><Link to="/edit">Edit Expense</Link></li>
            <li><Link to="/help">Help</Link></li>
        </ul>
    </header>
);

export default Header;