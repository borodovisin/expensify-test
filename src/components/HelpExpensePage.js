import React from 'react';

const HelpExpensePage = () => (
    <div>Help Expensive</div>
);

export default HelpExpensePage;