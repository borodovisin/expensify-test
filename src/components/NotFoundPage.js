import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
    <div align="center">
    404 - <Link to="/">Go to home</Link>
    </div>
);

export default NotFoundPage;