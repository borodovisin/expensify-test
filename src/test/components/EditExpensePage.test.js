import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../components/EditExpensePage';

let editExpense, removeExpense, history, wrapper;

beforeEach(() => {
    editExpense = jest.fn();
    removeExpense = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(<EditExpensePage editExpense={editExpense} removeExpense={removeExpense}  
        history={history} expense={{amount: 195, description: "Gum", note: ""}} />);
})

test('should render EditExpense', () => {
    expect(wrapper).toMatchSnapshot();
})

test('should handle onSubmit', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')({amount: 195, description: "Gum", note: ""});
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(editExpense).toHaveBeenLastCalledWith(undefined, {amount: 195, description: "Gum", note: ""});
})