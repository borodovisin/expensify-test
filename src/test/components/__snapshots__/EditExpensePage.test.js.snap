// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpense 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 195,
        "description": "Gum",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
